From cb343b6a1969f38af77f72630c2333baa250df97 Mon Sep 17 00:00:00 2001
From: Shuguang Gong <shuguang.gong@intel.com>
Date: Wed, 2 Jul 2014 16:51:46 +0800
Subject: [PATCH 223/230] atomisp2: modify padding and CSI config

The padding size should be set to zero for SoC sensor,
and CSI lanes can change for different resolution settings.

Change-Id: Ifddf529f0958a7ef83a19235eb8fff93ffcacdaa
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-15968
Signed-off-by: Shuguang Gong <shuguang.gong@intel.com>
Reviewed-on: https://android.intel.com:443/314864
Reviewed-on: https://icggerrit.ir.intel.com/35727
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c   | 1 +
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c       | 5 +++++
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c  | 2 +-
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.h  | 1 +
 4 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index cfb690a..3cf3c64 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -5083,6 +5083,7 @@ int atomisp_set_fmt(struct video_device *vdev, struct v4l2_format *f)
 		if (ret)
 			return -EINVAL;
 
+		atomisp_csi_lane_config(isp);
 		crop_needs_override = true;
 	}
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c
index de3f7b8..0f74b9e 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c
@@ -404,6 +404,11 @@ int atomisp_subdev_set_selection(struct v4l2_subdev *sd,
 			padding_h = 12;
 		}
 
+		if (isp->inputs[isp_sd->input_curr].type == SOC_CAMERA) {
+			padding_w = 0;
+			padding_h = 0;
+		}
+
 		if (atomisp_subdev_format_conversion(isp_sd,
 						     isp_sd->capture_pad)
 		    && crop[pad]->width && crop[pad]->height)
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
index 3b81bdd..26194f7 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
@@ -578,7 +578,7 @@ static int atomisp_resume(struct device *dev)
 }
 #endif
 
-static int atomisp_csi_lane_config(struct atomisp_device *isp)
+int atomisp_csi_lane_config(struct atomisp_device *isp)
 {
 	static const struct {
 		u8 code;
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.h
index 025fe03..0023e54 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.h
@@ -34,5 +34,6 @@ void atomisp_video_unregister(struct atomisp_video_pipe *video);
 int atomisp_video_register(struct atomisp_video_pipe *video,
 	struct v4l2_device *vdev);
 const struct firmware *atomisp_load_firmware(struct atomisp_device *isp);
+int atomisp_csi_lane_config(struct atomisp_device *isp);
 
 #endif /* __ATOMISP_V4L2_H__ */
-- 
1.9.1

